### PR TITLE
feat: removing natural language function names and comments

### PR DESCRIPTION
**Pull request by vdaita**
Tested it out with a Python file.
My current solution for dealing with empty comment lines (like "#" and "//") is to sort by length so that these appear last and don't affect the removal of longer comments. 
Moved tree-sitter queries to utility, since the comment queries were being used in both function_analysis and obfuscate_nl.

### CODE REVIEW

**Comment by vdaita** [comment]
@JialeTomTian This replaces the comments with empty bytes and saves it. The assert statements makes sure that the number of bytes are the same size, so the same line numbers and bytes should correspond. Does this work?

File: scripts/curate/utility.py [hunk 1]
```diff
@@ -11,3 +11,46 @@
     "php": [".php"],
     "rust": [".rs"],
 }
+
```
Comment (JialeTomTian) [inline scripts/curate/utility.py:14]: We have moved this to `repoqa.utility` you can use it directly from there.

**Review by JialeTomTian — COMMENTED** [review]
I think this approach should work, the only concern is that we need to remove all null characters when we construct the context in `search_needle_function`. The `code-llama tokenizer` will tokenize null characters as `<0x00>` which would mess up our prefix/suffix length calculation. @ganler thoughts?

**Review by JialeTomTian — APPROVED** [review]
Really great overall! Just made some minor changes. Thanks again for proposing this idea and also completing the implementation. We will run more experiments with this in the future and update the results!

**Pull request merged**